### DISCUSSION TITLE
Republishing Course goes to "Failed" status

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#665] [General Discussion]
SB Instance/version:  
 SB version 5.0 (Upgraded recently from 4.6 to 5.0)
SCENARIO
1. Open a Live course

1. edit the course
2. Send of review/publish
3. Reviewer logs in and publish the course

At this point the course is going to "Failed" status instead of "Live"

Debugging Result:
1. Noticed the following publish error

Debug: When we read the course using read api there we are seeing the issue as

```
"status": "Failed",
            "code": "org.sunbird.Z8oJxT.copy",
            "publishError": "key not found: do_1139422466574090241818",
            "credentials": {
                "enabled": "Yes"
            },
```

2. We understand there is a new field called relational_metadata got introduced and there is a mismatch of values of this field in cassandra and neo4j, this issue occur.    The only way to over come  this currently is to nullify this value in the databases  for the affected content and then republish which is not a very feasible solution 

Attaching the log for your reference.   

`content-publish-taskmanager-765d5945c8-rzk5g content-publish-taskmanager 2023-12-07T17:04:49.021905960+05:30 2023-12-07 11:34:49,021 INFO  org.sunbird.job.content.function.PublishEventRouter          [] - PublishEventRouter :: Event: JobRequest {map={actor=Map(id -> Publish Samza Job, type -> System), eid=BE_JOB_REQUEST, edata=Map(publish_type -> public, contentType -> Course, iteration -> 1, metadata -> Map(identifier -> do_1138821588798996481609, mimeType -> application/vnd.ekstep.content-collection, objectType -> Collection, lastPublishedBy -> 7b1f1cba-a162-410a-a5cf-f1a7f05feed9, pkgVersion -> 2), action -> publish), ets=1701948888871, context=Map(pdata -> Map(ver -> 1.0, id -> org.sunbird.platform), channel -> 0137236500887961602, env -> production), mid=LP.1701948888871.1900523b-fb64-49aa-8702-452e69454b71, object=Map(ver -> 1701948850112, id -> do_1138821588798996481609)}}
content-publish-taskmanager-765d5945c8-rzk5g content-publish-taskmanager 2023-12-07T17:04:49.022164200+05:30 2023-12-07 11:34:49,022 INFO  org.sunbird.job.content.function.PublishEventRouter          [] - PublishEventRouter :: Sending Collection For Publish Having Identifier: do_1138821588798996481609
content-publish-taskmanager-765d5945c8-rzk5g content-publish-taskmanager 2023-12-07T17:04:49.073457537+05:30 2023-12-07 11:34:49,073 INFO  org.sunbird.job.content.function.CollectionPublishFunction   [] - Collection publishing started for : do_1138821588798996481609
content-publish-taskmanager-765d5945c8-rzk5g content-publish-taskmanager 2023-12-07T17:04:49.074972668+05:30 2023-12-07 11:34:49,074 INFO  org.sunbird.job.publish.helpers.ObjectReader                 [] - Reading editable object data for: do_1138821588798996481609 with pkgVersion: 2.0
content-publish-taskmanager-765d5945c8-rzk5g content-publish-taskmanager 2023-12-07T17:04:49.087271147+05:30 2023-12-07 11:34:49,086 INFO  org.sunbird.job.publish.helpers.ObjectReader                 [] - ObjectReader:: getMetadata:: identifier: do_1138821588798996481609 with objType: CollectionImage
content-publish-taskmanager-765d5945c8-rzk5g content-publish-taskmanager 2023-12-07T17:04:49.087340636+05:30 2023-12-07 11:34:49,086 INFO  org.sunbird.job.publish.helpers.ObjectReader                 [] - Reading metadata for: do_1138821588798996481609 with metadata: Map(copyright -> DemoSidSetup, lastStatusChangedOn -> 2023-12-07T11:33:12.371+0000, publish_type -> public, author -> demoSiddhi Creater, se_mediumIds -> [demosid_medium_english], organisation -> [DemoSidSetup], mediaType -> content, name -> English, toc_url -> https://sunbirdsaaspublic.blob.core.windows.net/content/content/do_1138821588798996481609/artifact/do_1138821588798996481609_toc.json, batches -> [{"createdFor":["01386302081147699214"],"endDate":null,"name":"demo","batchId":"013911925124333568431","enrollmentType":"open","enrollmentEndDate":null,"startDate":"2023-10-25","status":1}], discussionForum -> {"enabled":"Yes"}, createdOn -> 2023-09-13T10:04:32.650+0000, createdFor -> [01386302081147699214], channel -> 0137236500887961602, generateDIALCodes -> Yes, lastUpdatedOn -> 2023-09-13T10:21:35.096+0000, subject -> [Science], size -> 4977, relational_metadata -> {"do_1138821588798996481609":{"name":"English","children":["do_1138821594326876161612"],"root":true},"do_1138821594326876161612":{"name":"Course Unit1","children":["do_1138821594326548481610"],"root":false},"do_1138821594326548481610":{"name":"learn1 ","children":["do_11372855875522560013","do_11394236450684928011053"],"root":false},"do_11372855875522560013":{"name":"Tekdi sachin 08 PDF","children":[],"root":false},"do_11394236450684928011053":{"name":"Course Unit 2","children":["do_113933078560186368134"],"root":false},"do_113933078560186368134":{"name":"Tekdi sachin PDF 2411","children":[],"root":false}}, targetMediumIds -> [demosid_medium_english], identifier -> do_1138821588798996481609.img, se_gradeLevelIds -> [demosid_gradelevel_grade11], description -> Enter description for Course, resourceType -> Course, ownershipType -> [createdBy], compatibilityLevel -> 4, targetBoardIds -> [demosid_board_demosid], audience -> [Student], trackable -> {"enabled":"Yes","autoBatch":"Yes"}, se_boards -> [demosid], os -> [All], primaryCategory -> Course, issuer -> {"name":"SaaS","url":"https://sunbirdsaas.com/"}, appIcon -> , SYS_INTERNAL_LAST_UPDATED_ON -> 2023-12-05T08:54:44.010+0000, se_mediums -> [Engish], se_subjectIds -> [demosid_subject_science], downloadUrl -> https://sunbirdsaaspublic.blob.core.windows.net/content/content/do_1138821588798996481609/english_1694600494971_do_1138821588798996481609_2_SPINE.ecar, se_subjects -> [Science], lockKey -> 74fb18df-573f-4fa0-8124-b45fef7bfb19, attributions -> [], framework -> demosid, creator -> demoSiddhi Creater, totalCompressedSize -> 4423916, versionKey -> 1701948850112, mimeType -> application/vnd.ekstep.content-collection, sYS_INTERNAL_LAST_UPDATED_ON -> 2023-09-13T10:21:34.971+0000, code -> org.sunbird.jU3CbB, se_boardIds -> [demosid_board_demosid], license -> CC BY 4.0, leafNodes -> [do_11372855875522560013], version -> 2, prevStatus -> Processing, contentType -> Course, prevState -> Review, language -> [English], lastPublishedOn -> 2023-09-13T10:21:34.934+0000, contentTypesCount -> {"Resource":1,"CourseUnit":2}, objectType -> CollectionImage, subjectIds -> [demosid_subject_science], status -> Review, reservedDialcodes -> {"R6M3V9":0}, targetFWIds -> [demosid], createdBy -> 5f77e496-8998-4e32-89dc-d963de9023e0, dialcodeRequired -> No, lastSubmittedOn -> 2023-12-07T11:34:10.098+0000, keywords -> [PDF], dialcodes -> [R6M3V9], userConsent -> Yes, idealScreenSize -> normal, contentEncoding -> gzip, leafNodesCount -> 1, depth -> 0, consumerId -> 5b7b73c2-4c29-4ebc-9f39-b415716ecf9f, lastPublishedBy -> 7b1f1cba-a162-410a-a5cf-f1a7f05feed9, targetSubjectIds -> [demosid_subject_science], mimeTypesCount -> {"application/pdf":1,"application/vnd.ekstep.content-collection":2}, se_gradeLevels -> [Grade 11], osId -> org.ekstep.quiz.app, copyrightYear -> 21, se_FWIds -> [demosid], appId -> production.production.portal, s3Key -> content/do_1138821588798996481609/artifact/do_1138821588798996481609_toc.json, contentDisposition -> inline, additionalCategories -> [], childNodes -> [do_11372855875522560013, do_1138821594326548481610, do_1138821594326876161612, do_113933078560186368134, do_11394236450684928011053], visibility -> Default, credentials -> {"enabled":"Yes"}, targetGradeLevelIds -> [demosid_gradelevel_grade11], variants -> {"spine":{"ecarUrl":"https://sunbirdsaaspublic.blob.core.windows.net/content/content/do_1138821588798996481609/english_1694600494971_do_1138821588798996481609_2_SPINE.ecar","size":"4977"},"online":{"ecarUrl":"https://sunbirdsaaspublic.blob.core.windows.net/content/content/do_1138821588798996481609/english_1694600495028_do_1138821588798996481609_2_ONLINE.ecar","size":"4977"}}, pkgVersion -> 2, idealScreenDensity -> hdpi)
content-publish-taskmanager-765d5945c8-rzk5g content-publish-taskmanager 2023-12-07T17:04:49.087378206+05:30 2023-12-07 11:34:49,086 INFO  org.sunbird.job.publish.helpers.ObjectReader                 [] - Reading extData for: do_1138821588798996481609 with extData: None
content-publish-taskmanager-765d5945c8-rzk5g content-publish-taskmanager 2023-12-07T17:04:49.113774356+05:30 2023-12-07 11:34:49,113 INFO  org.sunbird.job.publish.helpers.ObjectUpdater                [] - ObjectUpdater:: updateProcessingNode:: Query: MATCH (n:domain{IL_UNIQUE_ID:"do_1138821588798996481609.img"}) SET n.status="Processing",n.prevState="Review",n.copyright="DemoSidSetup",n.author="demoSiddhi Creater",n.se_mediumIds=["demosid_medium_english"],n.organisation=["DemoSidSetup"],n.mediaType="content",n.name="English",n.toc_url="https://sunbirdsaaspublic.blob.core.windows.net/content/content/do_1138821588798996481609/artifact/do_1138821588798996481609_toc.json",n.batches="[{\"createdFor\":[\"01386302081147699214\"],\"endDate\":null,\"name\":\"demo\",\"batchId\":\"013911925124333568431\",\"enrollmentType\":\"open\",\"enrollmentEndDate\":null,\"startDate\":\"2023-10-25\",\"status\":1}]",n.discussionForum="{\"enabled\":\"Yes\"}",n.createdOn="2023-09-13T10:04:32.650+0000",n.createdFor=["01386302081147699214"],n.channel="0137236500887961602",n.generateDIALCodes="Yes",n.subject=["Science"],n.size=4977,n.relational_metadata="{\"do_1138821588798996481609\":{\"name\":\"English\",\"children\":[\"do_1138821594326876161612\"],\"root\":true},\"do_1138821594326876161612\":{\"name\":\"Course Unit1\",\"children\":[\"do_1138821594326548481610\"],\"root\":false},\"do_1138821594326548481610\":{\"name\":\"learn1 \",\"children\":[\"do_11372855875522560013\",\"do_11394236450684928011053\"],\"root\":false},\"do_11372855875522560013\":{\"name\":\"Tekdi sachin 08 PDF\",\"children\":[],\"root\":false},\"do_11394236450684928011053\":{\"name\":\"Course Unit 2\",\"children\":[\"do_113933078560186368134\"],\"root\":false},\"do_113933078560186368134\":{\"name\":\"Tekdi sachin PDF 2411\",\"children\":[],\"root\":false}}",n.targetMediumIds=["demosid_medium_english"],n.se_gradeLevelIds=["demosid_gradelevel_grade11"],n.description="Enter description for Course",n.resourceType="Course",n.ownershipType=["createdBy"],n.compatibilityLevel=4,n.targetBoardIds=["demosid_board_demosid"],n.audience=["Student"],n.trackable="{\"enabled\":\"Yes\",\"autoBatch\":\"Yes\"}",n.se_boards=["demosid"],n.os=["All"],n.primaryCategory="Course",n.issuer="{\"name\":\"SaaS\",\"url\":\"https://sunbirdsaas.com/\"}",n.appIcon="",n.SYS_INTERNAL_LAST_UPDATED_ON="2023-12-05T08:54:44.010+0000",n.se_mediums=["Engish"],n.se_subjectIds=["demosid_subject_science"],n.downloadUrl="https://sunbirdsaaspublic.blob.core.windows.net/content/content/do_1138821588798996481609/english_1694600494971_do_1138821588798996481609_2_SPINE.ecar",n.se_subjects=["Science"],n.lockKey="74fb18df-573f-4fa0-8124-b45fef7bfb19",n.attributions=[],n.framework="demosid",n.creator="demoSiddhi Creater",n.totalCompressedSize=4423916,n.versionKey="1701948850112",n.mimeType="application/vnd.ekstep.content-collection",n.sYS_INTERNAL_LAST_UPDATED_ON="2023-09-13T10:21:34.971+0000",n.code="org.sunbird.jU3CbB",n.se_boardIds=["demosid_board_demosid"],n.license="CC BY 4.0",n.leafNodes=["do_11372855875522560013"],n.version=2,n.prevStatus="Processing",n.contentType="Course",n.prevState="Review",n.language=["English"],n.lastPublishedOn="2023-09-13T10:21:34.934+0000",n.contentTypesCount="{\"Resource\":1,\"CourseUnit\":2}",n.subjectIds=["demosid_subject_science"],n.reservedDialcodes="{\"R6M3V9\":0}",n...`

Is there any solution for this

**Comment by vinukumar-vs** [comment]
@siddhishinde0723 Anything related to Content management, I would suggest to have this discussion on [sunbird-knowlg building block](https://knowlg.sunbird.org/) for ease of access or get quick responses.

Can move this query to [sunbird-knowlg discussion ](https://github.com/orgs/Sunbird-Knowlg/discussions)

@anisha147 [@pallakartheekreddy](https://github.com/pallakartheekreddy)

**Reply by siddhishinde0723** [reply]
Moved to [sunbird-knowlg discussion](https://github.com/orgs/Sunbird-Knowlg/discussions)   here is the [link](https://github.com/orgs/Sunbird-Knowlg/discussions/174) for ticket
